framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceImplementation.java

```diff
@@ -222,21 +233,35 @@ private boolean isRecursiveCall(MethodInvocationNode methodInvNode) {
    *
    * @param methodElt the element of the method or constructor being invoked
    * @param arguments the arguments of the invocation
+   * @param receiver the receiver node, if there is one; null if there is not
    * @param invocationTree the method or constructor invocation, used to viewpoint adapt any
    *     dependent types when storing newly-inferred annotations
    */
   private void updateInferredExecutableParameterTypes(
-      ExecutableElement methodElt, List<Node> arguments, Tree invocationTree) {
-
-    // TODO: this method should be updated to:
-    // 1. take the receiver parameter of the method being invoked as an argument, in node form,
-    //    if there is one (the argument should be null otherwise)
-    // 2. infer types for the method declaration based on the type of the receiver
-    // 3. DependentTypesHelper#delocalizeAtCallsite should be updated to handle dependent types
-    //    that refer to the receiver (there is a (commented-out) test for this in
-    //    tests/ainfer-index/non-annotated/DependentTypesViewpointAdapationTest.java).
+      ExecutableElement methodElt,
+      List<Node> arguments,
+      @Nullable Node receiver,
+      ExpressionTree invocationTree) {
 
     String file = storage.getFileForElement(methodElt);
+    // TODO: why can a constructor possibly have a non-null receiver? That makes no sense, because
```

Comment:
A constructor for a nested non-static class (that is, an inner class) has a receiver that is a reference to the outer class instance that encloses the class.